runtime/oti/j9.h

```diff
@@ -273,6 +273,8 @@ static const struct { \
 #define MAXIMUM_HEAP_SIZE_RECOMMENDED_FOR_COMPRESSEDREFS	((U_64)57 * 1024 * 1024 * 1024)
 #define MAXIMUM_HEAP_SIZE_RECOMMENDED_FOR_3BIT_SHIFT_COMPRESSEDREFS	((U_64)25 * 1024 * 1024 * 1024)
 
+#define J9_NATIVE_FOOTPRINT_EXCLUDING_HEAP ((U_64)512 * 1024 * 1024)
```

Comment:
Given this is only used in GCExtensions.cpp, shouldn't it be defined there?
Where does this value come from?  It needs a comment describing why this value is valid at a minimum

Reply:
I can move this to GCExtensions.cpp.
This value is suggested by @mpirvu in his comment https://github.com/eclipse/openj9/issues/1429#issuecomment-375045690. I guess this is just a rough estimate, isn't it @mpirvu ? Not sure if there is a better way to estimate this value.

Reply:
The rationale from that comment should be included in the code so we can figure out where the magic numbers came from in the future.

Reply:
@DanHeidinga I updated the comment as:
```
                /* If running in a cgroup with memory limit > 1G, reserve at-least 512 MB for JVM's internal requirements
                 * like JIT compilation etc, and extend default max heap memory to at-most 75% of cgroup limit.
                 */
#define J9_NATIVE_FOOTPRINT_EXCLUDING_HEAP ((U_64)512 * 1024 * 1024)
```
Is that better?

Reply:
Should the #define include `cgroup` or `container` in its name?  

I'd prefer describing the purpose of this define before tying it to cgroups & heap sizes so that the validity of the constant is clear.  Maybe something like:
```C
/* A very rough estimate of the minimum amount of memory required by the JVM excluding
 * the heap.  The size is a conservative estimate, attempting to leave room the JVM's internal
 * requirements given that one compilation thread can use up to 256M.
 */
#define J9_NATIVE_FOOTPRINT_EXCLUDING_HEAP ((U_64)512 * 1024 * 1024)
```

The rest of your comment is useful info and should be included in `computeDefaultMaxHeap`.